src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckResponseTest.java

```diff
@@ -0,0 +1,191 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.io.Resource;
+import org.springframework.test.annotation.DirtiesContext;
+import uk.nhs.digital.nhsconnect.lab.results.IntegrationBaseTest;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.MeshMessage;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Instant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests that the correct NHSACK is returned for valid interchanges and when errors occur in processing.
+ * There is an NHSACK status code for each type of response:
+ *
+ * IAF: Valid Interchange, all Messages accepted
+ * IAP: Valid Interchange, some Messages accepted, some Messages rejected
+ * IRA: Valid Interchange, but all Messages were rejected
+ * IRM: Interchange rejected due to an error in one message
+ * IRI: Interchange rejected due to interchange level error
+ */
+@DirtiesContext
+class NhsAckResponseTest extends IntegrationBaseTest {
+    @Value("classpath:edifact/pathology_IAF.edifact.dat")
+    private Resource edifactIAFResource;
+
+    @Value("classpath:edifact/pathology_IAF.nhsack.dat")
+    private Resource nhsAckIAFResource;
+
+    @Value("classpath:edifact/pathology_IAP.edifact.dat")
+    private Resource edifactIAPResource;
+
+    @Value("classpath:edifact/pathology_IAP.nhsack.dat")
+    private Resource nhsAckIAPResource;
+
+    @Value("classpath:edifact/pathology_IRA.edifact.dat")
+    private Resource edifactIRAResource;
+
+    @Value("classpath:edifact/pathology_IRA.nhsack.dat")
+    private Resource nhsAckIRAResource;
+
+    @Value("classpath:edifact/pathology_IRM.edifact.dat")
+    private Resource edifactIRMResource;
+
+    @Value("classpath:edifact/pathology_IRM.nhsack.dat")
+    private Resource nhsAckIRMResource;
+
+    @Value("classpath:edifact/pathology_IRI.edifact.dat")
+    private Resource edifactIRIResource;
+
+    @Value("classpath:edifact/pathology_IRI.nhsack.dat")
+    private Resource nhsAckIRIResource;
+
+    @MockBean
+    private TimestampService timestampService;
+
+    @MockBean
+    private SequenceService sequenceService;
+
+    private static final Instant FIXED_TIME = Instant.parse("2020-04-27T16:37:00Z");
+    private static final Long FIXED_SEQUENCE_NUMBER = 1000L;
+
+    @BeforeEach
+    void setUp() {
+        when(timestampService.getCurrentTimestamp()).thenReturn(FIXED_TIME);
+        when(sequenceService.generateInterchangeSequence(any(), any())).thenReturn(FIXED_SEQUENCE_NUMBER);
+        clearGpOutboundQueue();
```

Comment:
We may also need to clear down the mesh outbound queue, as each test adds a message to this

Reply:
Added the clearMeshOutboundQueue method to setUp().